spec/controllers/accounts_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+
+RSpec.describe AccountsController, type: :controller do
+  let(:user) { build(:user) }
+  let!(:account) { create(:account, user: user) }
+
+  before do
+    allow_any_instance_of(described_class).to receive(:current_user).and_return(user)
+  end
+
+  context 'POST #create' do
+    context 'with valid attributes' do
+      it 'creates a new valid account' do
+        expect {
+          post :create, params: { account: attributes_for(:account) }
+        }.to change(Account, :count).by(1)
+      end
+
+      it 'redirects to the show account view' do
+        post :create, params: { account: attributes_for(:account) }
+        expect(response).to redirect_to(Account.last)
+      end
+    end
+
+    # context 'with invalid attributes' do
+    #   it 'does not create a new account' do
+    #     byebug
+    #     expect {
+    #       post :create, params: { account: attributes_for(:account, pin_wrong: true) }
+    #     }.to_not change(Account, :count)
+    #   end
+
+    #   it 'does not redirect to show account view' do
+    #     post :create, params: { account: attributes_for(:account, pin_wrong: true) }
+    #     expect(response).to render_template(:new)
+    #   end
+    # end
+  end
+
+  context 'PATCH #change_status' do
+    context 'when the account is active' do
+      before { account.update(status: :active) }
+
+      it 'changes the status to blocked' do
+        patch :change_status, params: { id: account.id }
+        expect(account.reload.status).to eq('blocked')
```

Comment:
why reload?

Reply:
the change status method updates the database record. To see the change on the account object I reloaded it.

Reply:
you don't have to do that